diff --git a/node_modules/@sap/ams/src/cap/env/amsPluginRuntime.js b/node_modules/@sap/ams/src/cap/env/amsPluginRuntime.js
index a48fc57..8993d1e 100644
--- a/node_modules/@sap/ams/src/cap/env/amsPluginRuntime.js
+++ b/node_modules/@sap/ams/src/cap/env/amsPluginRuntime.js
@@ -174,7 +174,8 @@ class AmsPluginRuntime {
 
 	/** Returns the path to the DCL files based on the cds env. A default value for *dclRoot* is supplied to cds env via the package.json. */
 	get dclRoot() {
-		return process.env.AMS_DCL_ROOT ?? this.amsCdsEnv.dclroot ?? this.amsCdsEnv.dclfolder ?? this.amsCdsEnv.dclRoot ?? this.amsCdsEnv.dclFolder;
+		const root = process.env.AMS_DCL_ROOT ?? this.amsCdsEnv.dclroot ?? this.amsCdsEnv.dclfolder ?? this.amsCdsEnv.dclRoot ?? this.amsCdsEnv.dclFolder;
+		return root && !path.isAbsolute(root) ? path.join(cds.root, root) : root;
 	}
 
 	/** Returns the name of the DCL package to which basePolicies.dcl is generated based on the cds env. A default value for *dclGenerationPackage* is supplied to cds env via the package.json. */
@@ -184,7 +185,8 @@ class AmsPluginRuntime {
 
 	/** Returns the path to the local DCN files based on the cds env.  A default value for *dcnRoot* is supplied to cds env via the package.json. */
 	get dcnRoot() {
-		return this.amsCdsEnv.dcnroot || this.amsCdsEnv.dcnRoot;
+		const root = this.amsCdsEnv.dcnroot || this.amsCdsEnv.dcnRoot;
+		return root && !path.isAbsolute(root) ? path.join(cds.root, root) : root;
 	}
 
 	/** Returns the URL of the AMS edge service based on the cds env, if configured. */
